gnomad_constraint/utils/constraint_basics.py

```diff
@@ -165,20 +167,221 @@ def get_proportion_observed_by_coverage(
+            - observed_variants_{pop} (where pop is each population) - observed variant counts for each population
+            - possible_variants - possible variant counts for each combination of 'context', 'ref', 'alt', 'methylation_level', 'mu_snp
+            - cpg - whether it's a cpg site or not
+            - mu_snp - mutation rate
+
+    :param ht: High coverage Table.
+    :param pop_anal: Whether to build models for each population. Defaults to True.
+    :param pops: List of populations. Defaults to `POPS`.
+    :param weighted: Whether to generalize the model to weighted least squares using 'possible_variants'.
+    :return: A Dictionary of intercepts and slopes for the full plateau models.
+    """
+    pop_lengths = get_all_pop_lengths(ht, pops)
+
+    plateau_models = build_plateau_models_total(ht, weighted=weighted)
+
+    if pop_anal:
+        arg_expr = {}
+        for length, pop in pop_lengths:
+            plateau_model = build_plateau_models_pop(ht, pop, length, weighted=weighted)
+            arg_expr[pop] = plateau_model[pop]
+        plateau_models = plateau_models.annotate(**arg_expr)
+
+    return plateau_models
+
+
+def build_plateau_models_pop(
+    ht: hl.Table, pop, length, weighted: bool = False
+) -> Dict[str, Tuple[float, float]]:
+    """
+    Calibrate high coverage model for each population in `POPS`.
+
+    :param ht: High coverage Table.
+    :param pop: List of populations. Defaults to `POPS`.
+    :param length: The indices of each population.
+    :param weighted: Whether to generalize the model to weighted least squares using 'possible_variants', defaults to False.
+    :return: A Dictionary of intercepts and slopes for plateau models of each population.
+    """
+    agg_expr = {
+        pop: [
+            hl.agg.group_by(
+                ht.cpg,
+                hl.agg.linreg(
+                    ht[f"observed_{pop}"][i] / ht.possible_variants,
+                    [1, ht.mu_snp],
+                    weight=ht.possible_variants if weighted else None,
+                ).beta,
+            )
+            for i in range(length)
+        ]
+    }
+    return ht.aggregate(hl.struct(**agg_expr))
+
+
+def build_plateau_models_total(
+    ht: hl.Table, weighted: bool = False
+) -> Dict[str, Tuple[float, float]]:
+    """
+    Calibrate high coverage model in total.
+
+    :param ht: High coverage Table.
+    :param weighted: Whether to generalize the model to weighted least squares using 'possible_variants', defaults to False.
+    :return: A Dictionary of intercepts and slopes for a plateau model.
+    """
+    agg_expr = {
+        "total": hl.agg.group_by(
+            ht.cpg,
+            hl.agg.linreg(
+                ht.observed_variants / ht.possible_variants,
+                [1, ht.mu_snp],
+                weight=ht.possible_variants if weighted else None,
+            ).beta,
+        )
+    }
+    return ht.aggregate(hl.struct(**agg_expr))
```

Comment:
same comment as below about passing in these field names as arguments rather than hardcoding